Code looks good, tests pass. When running locally, I see a lot of 

```
[ERROR] [12:46:24.110] [rawls-akka.actor.default-dispatcher-16] o.b.d.r.j.WorkflowSubmissionActor - Lock wait timeout exceeded; try restarting transaction
```

and I was able to generate a 500 NPE (perhaps unrelated) when trying to submit a 400 workflow submission and also editing the method in a separate tab. 

```
[ERROR] [05/24/2016 12:45:06.110] [rawls-akka.actor.default-dispatcher-11] [ActorSystem(rawls)] error processing request: http://app:8080/api/workspaces/broad-dsde-dev/grAddTest/methodconfigs/alex_methods/echo_strings
com.mysql.jdbc.exceptions.jdbc4.MySQLTransactionRollbackException: Lock wait timeout exceeded; try restarting transaction
    at sun.reflect.GeneratedConstructorAccessor97.newInstance(Unknown Source)
...
[ERROR] [05/24/2016 12:45:06.121] [rawls-akka.actor.default-dispatcher-11] [akka://rawls/user/rawls-service] Error during processing of request HttpRequest(PUT,http://app:8080/api/workspaces/broad-dsde-dev/grAddTest/methodconfigs/alex_methods/echo_strings,List(Content-Length: 791, Connection: Keep-Alive, X-Forwarded-Server: local.broadinstitute.org, X-Forwarded-Host: local.broadinstitute.org, X-Forwarded-For: 172.17.0.4, ... stuff removed ..., Host: app:8080),HttpEntity(application/json,{"name":"echo_strings","methodRepoMethod":{"methodNamespace":"alex_methods","methodName":"echo_strings","methodVersion":1},"outputs":{"echo_strings.echo_files.out":"\"Test\""},"inputs":{"echo_strings.echo_files.input10":"\"String\"","echo_strings.echo_files.input6":"\"String\"","echo_strings.echo_files.input9":"\"String\"","echo_strings.echo_files.input5":"\"String\"","echo_strings.echo_files.input3":"\"String\"","echo_strings.echo_files.input1":"\"String\"","echo_strings.echo_files.input7":"\"S...),HTTP/1.1)
java.lang.NullPointerException
    at spray.json.JsonPrinter$class.firstToBeEncoded$1(JsonPrinter.scala:58)
```

--- a/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala
+++ b/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala
@@ -160,10 +160,7 @@ trait AttributeComponent {
     }
 
     def batchInsertAttributes(attributes: Seq[RECORD]) = {
-      val recordsGrouped = attributes.grouped(batchSize).toSeq
-      DBIO.sequence(recordsGrouped map { batch  =>
-        (this ++= batch)
-      })
+      insertInBatches(this, attributes)
     }
 
     def marshalAttributeEntityReference(ownerId: OWNER_ID, name: String, listIndex: Option[Int], ref: AttributeEntityReference, entityIdsByRef: Map[AttributeEntityReference, Long]): RECORD = {

--- a/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/DriverComponent.scala
+++ b/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/DriverComponent.scala
@@ -71,4 +71,8 @@ trait DriverComponent {
     concatSqlActions(concatSqlActions(a, delim), b)
   }
 
+  def insertInBatches[R, T <: Table[R]](tableQuery: TableQuery[T], records: Seq[R]): WriteAction[Unit] = {
+    DBIO.seq(records.grouped(batchSize).map(tableQuery ++= _).toSeq:_*)
+  }
+
 }

--- a/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala
+++ b/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala
@@ -283,12 +283,11 @@ trait EntityComponent {
 
     def batchInsertEntities(workspaceContext: SlickWorkspaceContext, entities: Seq[EntityRecord]): ReadWriteAction[Seq[EntityRecord]] = {
       if(!entities.isEmpty) {
-        val recordsGrouped = entities.grouped(batchSize).toSeq
-        DBIO.sequence(recordsGrouped map { batch  =>
-          (entityQuery ++= batch)
-        })
-      } andThen selectEntityIds(workspaceContext, entities)
-      else DBIO.successful(Seq.empty[EntityRecord])
+        insertInBatches(entityQuery, entities) andThen selectEntityIds(workspaceContext, entities)
+      }
+      else {
+        DBIO.successful(Seq.empty[EntityRecord])
+      }
     }
 
     def selectEntityIds(workspaceContext: SlickWorkspaceContext, entities: Seq[EntityRecord]): ReadAction[Seq[EntityRecord]] = {

--- a/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala
+++ b/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala
@@ -148,7 +148,7 @@ trait WorkflowComponent {
           workflowEntityRec <- entityQuery if workflowEntityRec.id === workflowRec.workflowEntityId
         } yield (workflowRec, workflowEntityRec)
 
-        DBIO.seq(recsToInsert.grouped(batchSize).map(workflowQuery ++= _).toSeq:_*) andThen
+        insertInBatches(workflowQuery, recsToInsert) andThen
         insertedRecQuery.result.map(_.map { case (workflowRec, workflowEntityRec) =>
           AttributeEntityReference(workflowEntityRec.entityType, workflowEntityRec.name) -> workflowRec
         }.toMap)
@@ -168,7 +168,7 @@ trait WorkflowComponent {
           insertedInputResolutionRec <- submissionValidationQuery if insertedInputResolutionRec.workflowId === workflowRec.id
         } yield (workflowEntityRec, insertedInputResolutionRec)
 
-        DBIO.seq(inputResolutionRecs.grouped(batchSize).map(submissionValidationQuery ++= _).toSeq:_*) andThen
+        insertInBatches(submissionValidationQuery, inputResolutionRecs) andThen
         insertedRecQuery.result.map(_.map { case (workflowEntityRec, insertedInputResolutionRec) =>
           (AttributeEntityReference(workflowEntityRec.entityType, workflowEntityRec.name), insertedInputResolutionRec.inputName) -> insertedInputResolutionRec
         }.toMap)
@@ -190,7 +190,7 @@ trait WorkflowComponent {
           message <- workflow.messages
         } yield WorkflowMessageRecord(workflowRecsByEntity(workflow.workflowEntity).id, message.value)
 
-        DBIO.seq(messageRecs.grouped(batchSize).map(workflowMessageQuery ++= _).toSeq:_*)
+        insertInBatches(workflowMessageQuery, messageRecs)
       }
 
       val entitiesWithMultipleWorkflows = workflows.groupBy(_.workflowEntity).filter { case (entityRef, workflows) => workflows.size > 1 }.keys

